java/java.hints/src/org/netbeans/modules/java/hints/errors/EnablePreviewFeature.java

```diff
@@ -0,0 +1,415 @@
+                        if (!entry.getValue().contains(ENABLE_PREVIEW_FLAG + " ")) {
+                            properties.put(entry.getKey(), ENABLE_PREVIEW_FLAG + " " + entry.getValue());
+                        }
+                    }
+                }
+                if (mapp != null) {
+                    ModelHandle2.putMapping(mapp, prj, cfg);
+                }
+
+            }
+
+            return null;
+        }
+    }
+
+    private static void storeEditableProperties(final Project prj, final String propertiesPath, final EditableProperties ep)
+            throws IOException {
+        try {
+            ProjectManager.mutex().writeAccess(new Mutex.ExceptionAction<Void>() {
+                @Override
+                public Void run() throws IOException {
+                    FileObject propertiesFo = prj.getProjectDirectory().getFileObject(propertiesPath);
+                    if (propertiesFo != null) {
+                        OutputStream os = null;
+                        try {
+                            os = propertiesFo.getOutputStream();
+                            ep.store(os);
+                        } finally {
+                            if (os != null) {
+                                os.close();
+                            }
+                        }
+                    }
+                    return null;
+                }
+            });
+        } catch (MutexException ex) {
+        }
+    }
+
+    private static EditableProperties getEditableProperties(final Project prj, final String propertiesPath)
+            throws IOException {
+        try {
+            return ProjectManager.mutex().readAccess(new Mutex.ExceptionAction<EditableProperties>() {
+                @Override
+                public EditableProperties run() throws IOException {
+                    FileObject propertiesFo = prj.getProjectDirectory().getFileObject(propertiesPath);
+                    EditableProperties ep = null;
+                    if (propertiesFo != null) {
+                        InputStream is = null;
+                        ep = new EditableProperties(false);
+                        try {
+                            is = propertiesFo.getInputStream();
+                            ep.load(is);
+                        } finally {
+                            if (is != null) {
+                                is.close();
+                            }
+                        }
+                    }
+                    return ep;
+                }
+            });
+        } catch (MutexException ex) {
+            return null;
+        }
+    }
+
+    private boolean isMavenProject(Project prj) {
+
+        FileObject pom = prj.getProjectDirectory().getFileObject("pom.xml");
+        return (pom != null) && pom.isValid();
+
+    }
+
+    private boolean isAntProject(Project prj) {
+
+        FileObject buildFile = prj.getProjectDirectory().getFileObject("build.xml");
```

Comment:
This check feels way too weak: I'd suggest to at least check for nbproject/project.xml and nbproject/project.properties in addition to build.xml. Or, (if this is in a module that depends on project.ant), it might be possible to look for AntProjectHelper in the project's lookup.